### ISSUE TITLE
Distribution

### ISSUE THREAD

**Issue by Kyle873** [novafacing/selene#1]
First of all, this is a really helpful template, got me up and running very quickly, and can confirm that it works just fine on Windows while testing and debugging.

However, there is one thing that is giving me a major hang-up, and that is a proper fused distribution of the game. I've spent the better part of 2 hours messing around with package paths and `love.filesystem.setRequirePath` but have made basically no headway. I've gotten things to the point that it's able to find moonscript and lulpeg, but no matter what I do, it just refuses to find the main moon file when requiring it from main.lua. I could use some guidance on this, as this is my first time really messing around with lua's package paths and I honestly have no clue what I am doing.

```lua
local path = "?.lua;src/?.lua;src/?.moon;/lib/moonscript/?.lua;/lib/lulpeg/?.lua"
love.filesystem.setRequirePath(path)
package.path = package.path .. ";./lib/moonscript/?.lua"
package.path = package.path .. ";./lib/lulpeg/?.lua"
```

This is what I have currently ended up with and works fine for testing, but not for fusing.

I have a batch file that I use for building the fused executable:

```batch
@echo off

set NAME=Test
set PATH="C:\Program Files\7-Zip"
set LOVEPATH="C:\Program Files\LOVE"

mkdir bin
7z a -tzip %NAME%.love *.lua assets lib src
copy /b %LOVEPATH%\love.exe+%NAME%.love bin\%NAME%.exe
copy /b %LOVEPATH%\*.dll bin
del %NAME%.love

pause >nul
```

And the error:

```
Error

main.lua:1: module 'src.main' not found:
no field package.preload['src.main']
no 'src/main' in LOVE game directories.
no file 'src/main' in LOVE paths.
no file '.\src\main.lua'
no file 'C:\Projects\KTest\bin\lua\src\main.lua'
no file 'C:\Projects\KTest\bin\lua\src\main\init.lua'
no file './lib/moonscript/src\main.lua'
no file './lib/lulpeg/src\main.lua'
no file '.\src\main.dll'
no file 'C:\Projects\KTest\bin\src\main.dll'
no file 'C:\Projects\KTest\bin\loadall.dll'
no file '.\src.dll'
no file 'C:\Projects\KTest\bin\src.dll'
no file 'C:\Projects\KTest\bin\loadall.dll'


Traceback

[C]: in function 'require'
main.lua:1: in main chunk
[C]: in function 'require'
[C]: in function 'xpcall'
[C]: in function 'xpcall'
```

**Comment by Kyle873** [comment]
So I ended up deciding that the best course of action was to compile down the moon files into lua. It solved all my package path issues and seems like the "right" way to go about it for fusing/distribution. I would perhaps make a note of this in the README or some such for those who ran into this hitch such as myself, would rather others not bang their head against a wall for a few hours like I did 😄.

**Comment by novafacing** [comment]
Hmm, I can take a look at fixing this at some point, thanks for the test drive! I'll let you know if I get around to making changes that fix this, I usually just run my LOVE2D games from source so I didn't even run into the fusing problem yet :) not sure what's causing it as of right now.

Cheers!

**Comment by novafacing** [comment]
I think this is resolved by #5 @Kyle873 if you are still using this template for anything, it should be a lot easier now!
